@noun-auction/components/AuctionUi/SettleAuction/SettleAuctionBuilder.tsx

```diff
@@ -0,0 +1,65 @@
+import { useSigner } from 'wagmi'
+
+import { ContractTransaction } from 'ethers'
+
+import { useCallback, useEffect, useState } from 'react'
+
+import {
+  Auction as AuctionInterface,
+  Auction__factory as BuilderNounsAuction__factory,
+} from '@zoralabs/nouns-protocol/dist/typechain'
+
+import { SettleAuctionProps } from './SettleAuction'
+import { SettleAuctionComponent } from './SettleAuctionComponent'
+
+export function SettleAuctionBuiler({
+  useErrorMsg = false,
+  auctionContractAddress,
+  layout,
+  ...props
+}: SettleAuctionProps) {
+  const { data: signer } = useSigner()
+  const [isSuccess, setIsSuccess] = useState(false)
+  const [isLoading, setIsLoading] = useState(false)
+  const [isError, setIsError] = useState(false)
+  const [BuilderNounsAuction, setBuilderNounsAuction] = useState<AuctionInterface>()
+  const [txSubmitted, setTxSubmitted] = useState<ContractTransaction | undefined>(
+    undefined
+  )
+
+  useEffect(() => {
+    if (auctionContractAddress && signer) {
+      setBuilderNounsAuction(
+        BuilderNounsAuction__factory?.connect(auctionContractAddress, signer)
+      )
+    }
+  }, [auctionContractAddress, signer])
+
+  const handleOnSubmit = useCallback(
+    async (event) => {
```

Comment:
Likewise, I think ideally transactions like this would be handled in a hook rather than embedded within the component